irohad/ametsuchi/impl/burrow_storage.h

```diff
@@ -0,0 +1,61 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef IROHA_AMETSUCHI_BURROW_STORAGE_H
+#define IROHA_AMETSUCHI_BURROW_STORAGE_H
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+typedef struct {
+  char *data;
+  unsigned long long size;
+} Iroha_CharBuffer;
+
+typedef struct {
+  Iroha_CharBuffer *data;
+  unsigned long long size;
+} Iroha_CharBufferArray;
+
+typedef enum {
+  Iroha_Result_Type_Value,
+  Iroha_Result_Type_Error
+} Iroha_Result_Type;
+
+struct Iroha_Result {
+  Iroha_CharBuffer data;
+  Iroha_Result_Type which;
+};
+
+extern struct Iroha_Result Iroha_GetAccount(void *storage,
+                                            Iroha_CharBuffer address);
+
+extern struct Iroha_Result Iroha_UpdateAccount(void *storage,
+                                               Iroha_CharBuffer address,
+                                               Iroha_CharBuffer account);
+
+extern struct Iroha_Result Iroha_RemoveAccount(void *storage,
+                                               Iroha_CharBuffer address);
+
+extern struct Iroha_Result Iroha_GetStorage(void *storage,
+                                            Iroha_CharBuffer address,
+                                            Iroha_CharBuffer key);
+
+extern struct Iroha_Result Iroha_SetStorage(void *storage,
+                                            Iroha_CharBuffer address,
+                                            Iroha_CharBuffer key,
+                                            Iroha_CharBuffer value);
+
+extern struct Iroha_Result Iroha_StoreTxReceipt(void *storage,
```

Comment:
`StoreTxReceipt` is a bit misleading, because it is not a complete receipt, but only one `Log` call